src/Raven.Studio/typescript/viewmodels/database/advanced/databaseIDs.ts

```diff
@@ -0,0 +1,127 @@
+import appUrl = require("common/appUrl");
+import viewModelBase = require("viewmodels/viewModelBase");
+import accessManager = require("common/shell/accessManager");
+import getDatabaseDetailedStatsCommand = require("commands/resources/getDatabaseDetailedStatsCommand");
+import getDatabaseRecordCommand = require("commands/resources/getDatabaseRecordCommand");
+import saveUnusedDatabaseIDsCommand = require("commands/database/settings/saveUnusedDatabaseIDsCommand");
+import changeVectorUtils = require("common/changeVectorUtils");
+
+class databaseIDs extends viewModelBase {
+
+    isForbidden = ko.observable<boolean>(false);
+    
+    databaseID = ko.observable<string>();
+    databaseChangeVector = ko.observableArray<string>([]);
+    
+    unusedDatabaseIDs = ko.observableArray<string>([]);
+    inputDatabaseId = ko.observable<string>();
+
+    isSaveEnabled = ko.observable<boolean>();
+
+    spinners = {
+        save: ko.observable<boolean>(false)
+    };
+
+    constructor() {
+        super();
+        this.bindToCurrentInstance("addToUnusedList", "removeFromUnusedList");
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+        this.setupDisableReasons();
+    }
+    
+    canAddIdToUnusedIDs(cvEntry: string) {
+       return ko.pureComputed(() => changeVectorUtils.getDatabaseID(cvEntry) !== this.databaseID());
+    }
+
+    itemIsInsideUnusedList(cvEntry: string) {
+        return ko.pureComputed(() => {
+            const idPart = changeVectorUtils.getDatabaseID(cvEntry);
+            return _.includes(this.unusedDatabaseIDs(), idPart);
+        });
+    }
+
+    canActivate(args: any) {
+        return $.when<any>(super.canActivate(args))
+            .then(() => {
+                const deferred = $.Deferred<canActivateResultDto>();
+
+                this.isForbidden(!accessManager.default.operatorAndAbove());
+                
+                if (this.isForbidden()) {
+                    deferred.resolve({ can: true });
+                } else {
+                    const fetchStatsTask = this.fetchStats();
+                    const fetchUnusedIDsTask = this.fetchUnusedDatabaseIDs();
+
+                    return $.when<any>(fetchStatsTask, fetchUnusedIDsTask)
+                        .then(() => (deferred.resolve({ can: true })))
```

Comment:
remove ( ) around deferred no need for extra braces

Reply:
done